gen/k5login_generic

```diff
@@ -23,24 +22,18 @@ our $A_RESOURCE_K5LOGIN_TARGET_USER;  *A_RESOURCE_K5LOGIN_TARGET_USER =     \'ur
 my $k5loginDirectory = "$DIRECTORY/k5login";
 mkdir $k5loginDirectory or die "k5login directory can't be created: $!";
 
-#resource data
-my @resourcesData = $data->getChildElements;
 my $fileStructureWithData;
 #for each resource (k5login file)
-foreach my $resourceData (@resourcesData) {
-	#get resource attributes
-	my %resourceAttributes = attributesToHash $resourceData->getAttributes;
+foreach my $resourceId ($data->getResourceIds()) {
 	#get information about name of target user (file name for data)
-	my $fileName = $resourceAttributes{$A_RESOURCE_K5LOGIN_TARGET_USER};
+	my $fileName = $data->getResourceAttributeValue( resource => "$resourceId", attrName => "$A_RESOURCE_K5LOGIN_TARGET_USER" );
 	#prepare hash of users principals on resource
 	my %userPrincipals;
-	foreach my $memberData ($resourceData->getChildElements) {
-		#get members attributes
-		my %memberAttributes = attributesToHash $memberData->getAttributes;
+	foreach my $memberId ($data->getMemberIdsForResource( resource => "$resourceId" )) {
 		#if member is not valid, skip him
-		next if $memberAttributes{$A_USER_STATUS} ne 'VALID';
+		next if $data->getMemberAttributeValue( member => "$memberId", attrName => "$A_USER_STATUS" ) ne 'VALID';
 		#if member is valid, add value to hash
-		for my $principal (@{$memberAttributes{$A_PRINCIPALS}}) {
+		for my $principal (@{$data->getUserAttributeValue( member => "$memberId", attrName => "$A_PRINCIPALS" )}) {
 			$userPrincipals{$principal} = ();
 		}
 	}
```

Comment:
Please use `sort keys` in both `foreach` cycles below (when printing the structure to file). Output is then comparable and it also orders processed files.

Reply:
Added.